### PR TITLE
solo updates

### LINKED ISSUE #932: Additional features to enable solo CLI

### PR DESCRIPTION
**Pull request by njbernstein**
Adds early stopping by default to solo train

Adds `min_lr` option SCVI training

Closes #932

### CODE REVIEW

File: scvi/external/solo/_model.py [hunk 1]
```diff
@@ -174,6 +179,10 @@ def train(
             Minibatch size to use during training.
         plan_kwargs
             Keyword args for :class:`~scvi.lightning.ClassifierTrainingPlan`. Keyword arguments passed to
+        early_stopping_patience
```
Comment (adamgayoso) [inline scvi/external/solo/_model.py:182]: missing the doc for the `early_stopping` bool

File: scvi/external/solo/_model.py [hunk 2]
```diff
@@ -152,6 +153,10 @@ def train(
         validation_size: Optional[float] = None,
         batch_size: int = 128,
         plan_kwargs: Optional[dict] = None,
+        callbacks: list = [],
```
Comment (adamgayoso) [inline scvi/external/solo/_model.py:156]: shouldn't be necessary -- can be accessed through **kwargs

File: scvi/external/solo/_model.py [hunk 3]
```diff
@@ -185,13 +194,32 @@ def train(
             plan_kwargs.update(update_dict)
         else:
             plan_kwargs = update_dict
+
```
Comment (adamgayoso) [inline scvi/external/solo/_model.py:197]: check if callbacks is in **kwargs, if it isn't put an empty list there and then add the ES callback there. if it is, add ES to the existing list

File: scvi/lightning/_trainingplans.py [hunk 4]
```diff
@@ -69,6 +71,7 @@ def __init__(
         lr_scheduler_metric: Literal[
             "elbo_validation", "reconstruction_loss_validation", "kl_local_validation"
         ] = "elbo_validation",
+        lr_min: float = None,
```
Comment (adamgayoso) [inline scvi/lightning/_trainingplans.py:74]: default should be 0 like in pytorch right?

File: scvi/lightning/_trainingplans.py [hunk 5]
```diff
@@ -254,6 +261,7 @@ def __init__(
...
```
Comment (adamgayoso) [inline scvi/lightning/_trainingplans.py:264]: default of 0.0

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/lightning/_trainingplans.py (hunk 4 above)
Comment (njbernstein) [reply scvi/lightning/_trainingplans.py:74]: whoops ya

File: scvi/external/solo/_model.py (hunk 2 above)
Comment (njbernstein) [reply scvi/external/solo/_model.py:156]: ah sounds good

**Review by adamgayoso — APPROVED** [review]

File: scvi/external/solo/_model.py [hunk 6]
```diff
@@ -185,6 +195,28 @@ def train(
...
+                kwargs["callbacks"] += early_stopping_callback
+            else:
+                kwargs["callbacks"] = early_stopping_callback
+            kwargs["check_val_every_n_epoch"] = 1
+        else:
+            kwargs["check_val_every_n_epoch"] = (
+                kwargs["check_val_every_n_epoch"]
+                if kwargs["check_val_every_n_epoch"] is not None
+                else np.inf
+            )
```
Comment (adamgayoso) [inline scvi/external/solo/_model.py:218]:
```suggestion
```

**Pull request merged**